src/rlookup.h

```diff
@@ -199,29 +204,103 @@ typedef struct {
 
  * If F_OCREAT is not used, then this function will return NULL if a key could
  * not be found, unless OPT_UNRESOLVED_OK is set on the lookup itself. In this
  * case, the key is returned, but has the F_UNRESOLVED flag set.
  */
-RLookupKey *RLookup_GetKey(RLookup *lookup, const char *name, int flags);
+RLookupKey *RLookup_GetKey_TEMP(RLookup *lookup, const char *name, int flags);
 
 /**
  * Get or create a RLookup key for a given path and name. This function always returns a valid key,
  * hence, F_OCREAT and F_OEXCL are redundant here.
- * 
+ *
  * A key that contains a field from the index will be marked with F_SCHEMASRC.
- * 
+ *
  * This function first looks for an existing key with key->path equals to @path.
- * 
+ *
  * If this path is found, and @name doesn't equal @path, a new key is generated with the same
  * attributes as the found key, but with a different name.
- * 
+ *
+ */
+RLookupKey *RLookup_GetOrCreateKey_TEMP(RLookup *lookup, const char *path, const char *name, int flags);
+
+// End of old flags. Remove later and uncomment the following flags
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+typedef enum {
+  RLOOKUP_M_READ,   // Get key for reading (create oly if in schema and sortable)
+  RLOOKUP_M_WRITE,  // Get key for writing
+  RLOOKUP_M_LOAD,   // Load key from document (include known information on the key, fail if already loaded)
+} RLookupMode;
+
+// #define RLOOKUP_F_NOFLAGS 0x0 // No special flags to pass.
+
+/**
+ * This field is NOT part of the index schema, but created by a loader
+ * (we will look for it in the document)
  */
-RLookupKey *RLookup_GetOrCreateKey(RLookup *lookup, const char *path, const char *name, int flags);
+#define RLOOKUP_F_DOCSRC 0x01
+
+/**
+ * This field is part of the index schema.
+ */
+// #define RLOOKUP_F_SCHEMASRC 0x02
+
+/** Check the sorting table, if necessary, for the index of the key. */
+// #define RLOOKUP_F_SVSRC 0x04
+
+/**
+ * This key was created by the query itself (not in the document)
+ */
+#define RLOOKUP_F_QUERYSRC 0x08
+
+/** Copy the key string via strdup. `name` may be freed */
+// #define RLOOKUP_F_NAMEALLOC 0x10
+
+/**
+ * If the key is already present, then overwrite it (relevant only for LOAD or WRITE modes)
+ */
+#define RLOOKUP_F_OVERRIDE 0x20
+
+/**
+ * This key type is numeric
+ */
+#define RLOOKUP_T_NUMERIC 0x40
+
+/**
+ * This key is unresolved. It source needs to be derived from elsewhere
+ */
+// #define RLOOKUP_F_UNRESOLVED 0x80
+
+/**
+ * This field is hidden within the document and is only used as a transient
+ * field for another consumer. Don't output this field.
+ */
+// #define RLOOKUP_F_HIDDEN 0x100
+
+/**
+ * The opposite of F_HIDDEN. This field is specified as an explicit return in
+ * the RETURN list, so ensure that this gets emitted. Only set if
+ * explicitReturn is true in the aggregation request.
+ */
+// #define RLOOKUP_F_EXPLICITRETURN 0x200
```

Comment:
Not easily and I don't think we should try. A key that's marked with `RLOOKUP_F_HIDDEN` will never be returned, and a field that's marked with `RLOOKUP_F_EXPLICITRETURN` will always be returned. In both cases keys that aren't marked with any of them will be treated differently in different cases. This is easier to do instead of marking or unmrking all keys depends on the "global" mode to save one flag definition